### DISCUSSION TITLE
Action failing when workflow triggered by cron job?

### DISCUSSION THREAD

**Discussion by vincerubinetti** [stefanzweifel/git-auto-commit-action#160] [Help] [answered]
Hi there, thank you for the great tool. I wanted to talk about an issue I had, though I'm skeptical that it's a problem with the action, so I've done a discussion instead of an issue.

I've used this action in a [previous tool](https://github.com/greenelab/lab-website-template) without much difficulties or errors. That tool had the action in a workflow triggered on commit. Now I'm using the action in a [new tool](https://github.com/greenelab/preprint-bot) and getting intermittent errors. The new tool uses the action in a workflow triggered by a cron job, and I suspect that is the source of the issue.

Take a look at the action runs here: https://github.com/greenelab/preprint-bot/actions

Particularly, look at the three runs that resulted in errors 2 days ago. The first one was GitHub's fault I think, but the second one is  an issue with this action. Then the third error is a result of the previous one failing, as expected per my code.

Here is the error message related to this action:

<details><summary>error log</summary>

```
Started: bash /home/runner/work/_actions/stefanzweifel/git-auto-commit-action/v4/entrypoint.sh
INPUT_REPOSITORY value: .
INPUT_STATUS_OPTIONS: 
INPUT_BRANCH value: main
Already on 'main'
M	log.yaml
Your branch is up to date with 'origin/main'.
INPUT_ADD_OPTIONS: 
INPUT_FILE_PATTERN: log.yaml
INPUT_COMMIT_OPTIONS: 
INPUT_COMMIT_USER_NAME: GitHub Actions
INPUT_COMMIT_USER_EMAIL: actions@github.com
INPUT_COMMIT_MESSAGE: Run bot (comment mode) and update log
INPUT_COMMIT_AUTHOR: vincerubinetti <vincerubinetti@users.noreply.github.com>
[main 8823188] Run bot (comment mode) and update log
 Author: vincerubinetti <vincerubinetti@users.noreply.github.com>
 1 file changed, 86 insertions(+)
INPUT_TAGGING_MESSAGE: 
No tagging message supplied. No tag will be added.
INPUT_PUSH_OPTIONS: --force
remote: fatal error in commit_refs        
To https://github.com/greenelab/preprint-bot
 ! [remote rejected] HEAD -> main (failure)
error: failed to push some refs to 'https://github.com/greenelab/preprint-bot'
Error: Invalid status code: 1
    at ChildProcess.<anonymous> (/home/runner/work/_actions/stefanzweifel/git-auto-commit-action/v4/index.js:17:19)
    at ChildProcess.emit (events.js:210:5)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5) {
  code: 1
}
Error: Invalid status code: 1
    at ChildProcess.<anonymous> (/home/runner/work/_actions/stefanzweifel/git-auto-commit-action/v4/index.js:17:19)
    at ChildProcess.emit (events.js:210:5)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
```

</details>

All the other runs have worked though, which is odd. And it seems like more than a coincidence that the error came right after a GitHub-caused error.

Do you have any idea what could be going on here? I'm setting `--force` and `branch: main`, which I had to add due to earlier errors that have now been permanently resolved.

**Comment by stefanzweifel** [comment, marked as answer]
My gut instinct tells me, that this is related to [an outage](https://www.githubstatus.com/incidents/zbpwygxwb3gw) GitHub Actions had on the weekend.

The [run](https://github.com/greenelab/preprint-bot/actions/runs/847321795) you mentioned was executed on 16:24 UTC. The outage was marked as resolved on 17:05 UTC.

And the error message indicates that something was blowing up on GitHubs end.

```
remote: fatal error in commit_refs
```

Not much we can do here I'm afraid.
